platform/pulpcore/tasking/tasks.py

```diff
@@ -243,14 +245,26 @@ def apply_async(self, tags=[], group_id=None, args=None, kwargs=None, **options)
 
     def __call__(self, *args, **kwargs):
         """
-        Set the :class:`pulpcore.app.models.Task` object in the running state and log some output.
+        Called on the worker just before the task begin executing.
+
+        Set the :class:`pulpcore.app.models.Task` object in the running state,
+        associate the worker, and log a message.
 
         Skip the status updating if the task is called synchronously.
         """
-        if not self.request.called_directly:
-            task_status = TaskStatus.objects.get(pk=self.request.id)
-            task_status.set_running()
-        _logger.debug("Running task : [%s]" % self.request.id)
+        task_status = TaskStatus.objects.get(pk=self.request.id)
+        task_status.set_running()
+
+        try:
+            worker = Worker.objects.get(name=self.request.hostname)
+        except Worker.DoesNotExist:
+            _logger.warning(_("Failed to find worker with hostname {host}").format(
+                host=self.request.hostname))
+        else:
+            if not task_status.worker:
+                task_status.set_worker(worker)
```

Comment:
Can set_worker() be removed entirely and have it's two lines of function be moved here instead?